File: templates/scala/TemplateTest.scala
package nl.utwente.bigdata;
 
import org.apache.spark.rdd.RDD  
import org.apache.spark.{SparkConf, SparkContext}  
import org.scalatest.{BeforeAndAfter, FlatSpec, Matchers}

class ${toolName}Test extends FlatSpec with Matchers with BeforeAndAfter {

  var sc:SparkContext = _

  before {
    // create spark context
    val sparkConf = new SparkConf()
      .setMaster("local[*]")
      .setAppName("test")
    sc = new SparkContext(sparkConf)
  }

  after {
    sc.stop()
  }

  /* write similar test cases as the one above
  
  behavior of "Test"

  it should "count words in a text" in {
    // prepare input and expected input
    val text =
      """Hello Spark
        |Hello world
      """.stripMargin
    // convert input into rdd
    val input: RDD[String] = sc.parallelize(List(text))
    // calculate output
    val output: RDD[(String, Int)] = $toolName.doJob(input)
    // check output
    output.collect() should contain allOf (("Hello", 2), ("Spark", 1), ("world", 1))
  }
  */
}

File: templates/python/TemplateTest.py
import $toolName
import pytest
import pyspark

# create necessary test framework
@pytest.fixture(scope="session",
                params=[pytest.mark.spark_local('local'),
                        pytest.mark.spark_yarn('yarn')])
def spark_context(request):
    if request.param == 'local':
        conf = (pyspark.SparkConf()
                .setMaster("local[2]")
                .setAppName("pytest-pyspark-local-testing")
                )
    elif request.param == 'yarn':
        conf = (pyspark.SparkConf()
                .setMaster("yarn-client")
                .setAppName("pytest-pyspark-yarn-testing")
                .set("spark.executor.memory", "1g")
                .set("spark.executor.instances", 2)
                )
    request.addfinalizer(lambda: sc.stop())

    sc = pyspark.SparkContext(conf=conf)
    return sc

# add test cases
def test_that_requires_sc(spark_context):
  assert true
  #assert PythonWordCount.doJob(spark_context.parallelize(["12123 123123","3423"])).count() == 3